Task2/app/src/main/java/com/example/task2/MainActivity.kt

```diff
@@ -0,0 +1,110 @@
+package com.example.task2
+
+import android.os.Bundle
+import android.os.Handler
+import androidx.activity.ComponentActivity
+import android.widget.TextView
+import com.example.task2.ui.theme.Task2Theme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+
+class MainActivity : ComponentActivity() {
+
+    /* Time variables */
+    private var startTimeMillis: Long = 0
+    private var foregroundStartTimeMillis: Long = 0
+    private var backgroundStartTimeMillis: Long = 0
+    private var totalForegroundTimeMillis: Long = 0
+    private var totalBackgroundTimeMillis: Long = 0
+
+    /* To manage layouts */
+    private lateinit var foregroundTimeTextView: TextView
+    private lateinit var backgroundTimeTextView: TextView
+
+    private val handler = Handler()
+
+    private val updateForegroundTimeTask = object : Runnable {
+        override fun run() {
+            val elapsedTimeMillis = totalForegroundTimeMillis + (System.currentTimeMillis() - foregroundStartTimeMillis)
+            val elapsedSeconds = elapsedTimeMillis / 1000
+            val elapsedMinutes = elapsedSeconds / 60
+            val seconds = elapsedSeconds % 60
+            val milliseconds = elapsedTimeMillis % 1000
+            foregroundTimeTextView.text = String.format("ForegroundTime: %02d:%02d:%03d", elapsedMinutes, seconds, milliseconds)
```

Comment:
code to format the time should be extracted in a separate function

Reply:
Fixed!

Reply:
This was already reviewed in Task2, same case as below comment. However it was missing too in task3 project so I already corrected it too.